src/daft-core/src/utils/sketch.rs

```diff
@@ -0,0 +1,51 @@
+use common_error::DaftResult;
+use sketches_ddsketch::{Config, DDSketch};
+
+pub struct Sketch {
+    sketch: DDSketch,
+}
+
+impl Sketch {
+    pub fn new() -> Self {
+        Sketch {
+            sketch: DDSketch::new(Config::defaults()),
+        }
+    }
+
+    pub fn add(&mut self, value: f64) -> &mut Self {
+        self.sketch.add(value);
+        self
+    }
+
+    pub fn merge(&mut self, other: &Sketch) -> DaftResult<&mut Self> {
+        self.sketch.merge(&other.sketch)?;
+        Ok(self)
+    }
+
+    pub fn quantile(&self, q: f64) -> DaftResult<Option<f64>> {
+        Ok(self.sketch.quantile(q)?)
+    }
+
+    pub fn from_value(value: f64) -> Self {
```

Comment:
I think having `Sketch::new()` followed by an `add` is fine since that is the access pattern in most use-cases! (No need for `from_value`)